elide-standalone/src/main/java/com/yahoo/elide/standalone/interfaces/UserExtractionFunctionProvider.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2017, Oath Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.standalone.interfaces;
+
+import com.yahoo.elide.resources.DefaultOpaqueUserFunction;
+
+/**
+ * User extraction function provider.
+ */
+public interface UserExtractionFunctionProvider {
```

Comment:
I wonder if example would be simpler if we just directly used elide builder rather than di to setup elide?

Reply:
We do have this capability. We have 2 modes. In short, you have mechanisms to use ElideBuilder (full customizability, but need to setup everything still), or you give us a list of your security checks and user extraction function and we set everything else up with Hibernate5.